### PR TITLE
added length, params, and entropy for TuringDiagNormal

### PR DESCRIPTION
**Pull request by torfjelde**
Adds implementations of the following methods
- `Distributions.length(d::TuringDiagNormal)` (following Distributions.jl's doc, this is assumed to exist for any `MultivariateDistribution`)
- `Distributions.params(d::TuringDiagNormal)`
- `StatsBase.entropy(d::TuringDiagNormal)`

### CODE REVIEW

File: src/multivariate.jl [hunk 1]
```diff
@@ -55,6 +57,12 @@ function _logpdf(d::MvNormal, x::Union{Tracker.TrackedVector, Tracker.TrackedMat
     _logpdf(TuringMvNormal(d.μ, getchol(d.Σ)), x)
 end
 
+import StatsBase: entropy
+function entropy(d::TuringDiagNormal)
```
Comment (mohamed82008) [inline src/multivariate.jl]: Can we add a test for this?

Comment (torfjelde) [reply src/multivariate.jl]: Will do!

Comment (torfjelde) [reply src/multivariate.jl]: Sorry, completely forgot about this! Done diddit now though:)

**Comment by mohamed82008** [comment]
Thanks @torfjelde !

**Comment by torfjelde** [comment]
Oh crap, it seems I forgot to add `StatsBase` as to `[compat]` in Project.toml

**Pull request merged**